itest/litd_custom_channels_test.go

```diff
@@ -2046,6 +2051,176 @@ func testCustomChannelsLiquidityEdgeCases(_ context.Context,
 
 	logBalance(t.t, nodes, assetID, "after failed payment (asset "+
 		"invoice, strict forwarding)")
+
+	// Edge case: Check if the RFQ HTLC tracking accounts for cancelled
+	// htlcs. We achieve this by manually creating & using an RFQ quote with
+	// a set max amount. We first pay to a hodl invoice that we eventually
+	// cancel, then pay to a normal invoice which should succeed.
+
+	// We start by sloshing some funds in the Erin<->Fabia
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, 100_000, assetID,
+		fn.None[int64](), lnrpc.Payment_SUCCEEDED,
+		fn.None[lnrpc.PaymentFailureReason](),
+	)
+	sendKeySendPayment(t.t, erin, fabia, 20_000)
+
+	logBalance(t.t, nodes, assetID, "balance after 1st slosh")
+
+	// We create the RFQ order. We set the max amt to ~180k sats which is
+	// going to evaluate to about 10k assets.
+	resQ, err := charlieTap.RfqClient.AddAssetSellOrder(
+		ctxb, &rfqrpc.AddAssetSellOrderRequest{
+			AssetSpecifier: &rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetId{
+					AssetId: assetID,
+				},
+			},
+			PaymentMaxAmt:  180_000_000,
+			Expiry:         uint64(time.Now().Add(time.Hour).Unix()),
+			PeerPubKey:     dave.PubKey[:],
+			TimeoutSeconds: 100,
+		},
+	)
+	require.NoError(t.t, err)
+
+	// We now create a hodl invoice on Fabia, for 10k assets.
+	hodlInv := createAssetHodlInvoice(t.t, erin, fabia, 10_000, assetID)
+
+	// Charlie tries to pay via Dave, by providing the scid that was
+	// manually created above.
+	scid := rfqmsg.SerialisedScid(resQ.GetAcceptedQuote().Scid)
+	payInvoiceWithAssets(
+		t.t, charlie, dave, hodlInv.payReq, assetID, true,
+		fn.Some(lnrpc.Payment_IN_FLIGHT), fn.Some(scid),
+	)
+
+	time.Sleep(time.Second)
```

Comment:
No need to manually sleep here I think. `assertNumHtlcs` will wait in a loop until the required number is reached.